manifests/config.pp

```diff
@@ -158,128 +158,161 @@
+          $sssd_services = join(unique(pick($sssd['services'], []) + ['ifp']), ', ')
+          $sssd_ldap_user_extra_attrs = join(unique(pick($sssd['ldap_user_extra_attrs'], []) + ['email:mail', 'lastname:sn', 'firstname:givenname']), ', ')
+          $sssd_allowed_uids = join(unique(pick($sssd['allowed_uids'], []) + [$apache::user, 'root']), ', ')
+          $sssd_user_attributes = join(unique(pick($sssd['user_attributes'], []) + ['+email', '+firstname', '+lastname']), ', ')
+          $sssd_ifp_extra_attributes = [
+            "set target[.=~regexp('domain/.*')]/ldap_user_extra_attrs '${sssd_ldap_user_extra_attrs}'",
+            "set target[.='sssd']/services '${sssd_services}'",
+            'set target[.=\'ifp\'] \'ifp\'',
+            "set target[.='ifp']/allowed_uids '${sssd_allowed_uids}'",
+            "set target[.='ifp']/user_attributes '${sssd_user_attributes}'",
+          ]
+
+          $sssd_changes = $sssd_ifp_extra_attributes + ($foreman::ipa_sssd_default_realm ? {
+              undef => [],
+              default => ["set target[.='sssd']/default_domain_suffix '${$foreman::ipa_sssd_default_realm}'"],
+          })
+
+          augeas { 'sssd-ifp-extra-attributes':
+            context => '/files/etc/sssd/sssd.conf',
+            changes => $sssd_changes,
+            notify  => Service['sssd'],
+          }
+        }
+
+        foreman::settings_fragment { 'authorize_login_delegation.yaml':
+          content => template('foreman/settings-external-auth.yaml.erb'),
+          order   => '02',
         }
-      }
 
-      foreman::settings_fragment { 'authorize_login_delegation.yaml':
-        content => template('foreman/settings-external-auth.yaml.erb'),
-        order   => '02',
+        foreman::settings_fragment { 'authorize_login_delegation_api.yaml':
+          content => template('foreman/settings-external-auth-api.yaml.erb'),
+          order   => '03',
+        }
       }
+      'keycloak': {
+        $foreman_socket_override = undef
+
+        unless $foreman::ssl {
+          fail('Keycloak requires HTTPS')
+        }
 
-      foreman::settings_fragment { 'authorize_login_delegation_api.yaml':
-        content => template('foreman/settings-external-auth-api.yaml.erb'),
-        order   => '03',
+        foreman::settings_fragment { 'authorize_login_delegation.yaml':
+          content => template('foreman/settings-external-auth.yaml.erb'),
+          order   => '02',
+        }
+
+        # TODO: parameter
+        $keycloak_url = 'https://keycloak.example.com'
+        $oidc_issuer = "${keycloak_url}/auth/realms/${foreman::keycloak_realm}"
+        $keycloak_settings = {
+          ':login_delegation_logout_url' => "${foreman::foreman_url}/users/extlogout",
+          # TODO: parameters or obtain from ${oidc_issuer}/.well-known/openid-configuration
+          ':oidc_algorithm'              => 'RS256',
+          ':oidc_audience'               => ["${foreman::servername}-foreman-openidc"],
```

Comment:
I would parameterize this so that the value can be overwritten. This allows people who have a certain naming standards for their keycloak clients to keep using them.

Reply:
I agree, which is why there's a TODO above it.

Reply:
Apologies! Completely over looked those